inference-engine/src/inference_engine/cnn_network_ngraph_impl.cpp

```diff
@@ -355,6 +357,7 @@ CNNNetworkNGraphImpl::reshape(const std::map<std::string, std::vector<size_t>>&
                 ::ngraph::pass::Manager manager;
                 // resolves dynamism by replacing dynamic operation with static version
                 manager.register_pass<::ngraph::pass::ConvertNMS5ToLegacyMatcher>(false);
+                manager.register_pass<::ngraph::pass::DisableConvertConstantFoldingOnConstPath>();
```

Comment:
Do we really need to call this path in order to get output shapes?

Reply:
> Do we need to call this pass in order to get output shapes?
Yes, we need this pass to avoid unnecessary constant folding on weights (Constant -> Convert -> Subtract -> Multiply)
> Do we have tests which show that it is really needed?
We have tests for bot transformations: `DisableConvertConstantFoldingOnConstPath` and `ConstantFolding`. Performance validation should show if we have read time improvement (still in progress).

Reply:
If we disable constant folding on weight can we face with topologies where we will have dynamic output shapes without this constant folding?

Reply:
@ilyachur this is not the case. Please see how the transformation works first https://github.com/openvinotoolkit/openvino/blob/6d12d3c519b214588e8ece850ca6901a536af5c4/inference-engine/src/transformations/src/transformations/low_precision/disable_convert_constant_folding_on_const_path.cpp